plugin/core/types.py

```diff
@@ -127,18 +179,46 @@ def sel(self) -> 'Any':
     def score_selector(self, region: 'Any', scope: str) -> int:
         ...
 
+    def assign_syntax(self, syntax: str) -> None:
+        ...
+
+    def set_read_only(self, val: bool) -> None:
+        ...
+
+    def run_command(self, command_name: str, command_args: 'Optional[Dict[str, Any]]' = None) -> None:
+        ...
+
+    def find_all(self, selector: str) -> 'Iterable[Tuple[int, int]]':
+        ...
+
+    def add_regions(self, key: str, regions: 'Iterable[Any]', scope: str = "", icon: str = "", flags: int = 0) -> None:
+        ...
+
 
 class WindowLike(Protocol):
     def id(self) -> int:
         ...
 
+    def active_view(self) -> 'Optional[Any]':
+        ...
+
+    def run_command(self, cmd: str, args: 'Optional[Dict[str, Any]]') -> None:
+        ...
+
+    def new_file(self, flags: int, syntax: str) -> 'Any':
+        ...
+
+    def open_file(self, fname: str, flags: int, group: int) -> 'Any':
+        ...
+
     def is_valid(self) -> bool:
         ...
 
     def folders(self) -> 'List[str]':
         ...
 
-    def find_open_file(self, path: str) -> 'Optional[ViewLike]':
+    # should return Optional[ViewLike], but creates conflict when a real sublime.View is presented
+    def find_open_file(self, path: str) -> 'Optional[Any]':
```

Comment:
This for example will make mypy complain when the return type is an Optional[ViewLike]